react-components/src/components/Reveal3DResources/utils.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import {
+  type AddModelOptions,
+  type ClassicDataSourceType,
+  type CognitePointCloudModel,
+  type DataSourceType,
+  isDMPointCloudModel,
+  isClassicPointCloudModel
+} from '@cognite/reveal';
+import { useMemo } from 'react';
+import { isDMIdentifier } from './typeGuards';
+import { type PointCloudModelOptions, type CadPointCloudModelWithModelIdRevisionId } from './types';
+
+type MatchedModel = {
+  viewerModel: CognitePointCloudModel<DataSourceType>;
+  model: CadPointCloudModelWithModelIdRevisionId;
+};
+
+export function useModelsWithModelIdAndRevision(
+  models: PointCloudModelOptions[],
+  classicModelOptions: Array<AddModelOptions<ClassicDataSourceType>>
+): CadPointCloudModelWithModelIdRevisionId[] {
+  return useMemo(() => {
+    return classicModelOptions.map((model, index) => ({
+      modelOptions: models[index],
+      ...model
+    }));
+  }, [classicModelOptions, models]);
+}
+
+export function useMatchedModels(
+  viewerModels: Array<CognitePointCloudModel<DataSourceType>>,
+  modelsWithModelIdAndRevision: CadPointCloudModelWithModelIdRevisionId[]
+): MatchedModel[] {
```

Comment:
Since this seems to only operate with point cloud models, should the return type reflect that?